Makefile

```diff
@@ -1,42 +1,45 @@
-.PHONY: test build-test-container build push gotest gobuild
+.PHONY: build-test-container test-with-compose build push gotest gobuild down
 
+COMPOSE_FILE ?= tests/compose.yml
+CONTAINER_ENGINE ?= $(shell command -v podman > /dev/null 2>&1 && echo podman || echo docker )
+CONTAINER_SELINUX_FLAG ?= :z
 IMAGE_NAME := quay.io/app-sre/vault-manager
 IMAGE_TAG := $(shell git rev-parse --short=7 HEAD)
-DOCKER_CONF := $(CURDIR)/.docker
 GOOS := $(shell go env GOOS)
 PWD := $(shell pwd)
 
+ifneq (,$(wildcard $(CURDIR)/.docker))
+	DOCKER_CONF := $(CURDIR)/.docker
+else
+	DOCKER_CONF := $(HOME)/.docker
+endif
+
 gotest:
 	CGO_ENABLED=0 GOOS=$(GOOS) go test ./...
 
 gobuild: gotest
 	CGO_ENABLED=0 GOOS=$(GOOS) go build -a -buildvcs=false -installsuffix cgo ./cmd/vault-manager
 
 build:
-	@docker build --no-cache -t $(IMAGE_NAME):$(IMAGE_TAG) .
+	@$(CONTAINER_ENGINE) build --no-cache -t $(IMAGE_NAME):$(IMAGE_TAG) .
 
 push:
-	@docker tag $(IMAGE_NAME):$(IMAGE_TAG) $(IMAGE_NAME):latest
-	@docker --config=$(DOCKER_CONF) push $(IMAGE_NAME):$(IMAGE_TAG)
-	@docker --config=$(DOCKER_CONF) push $(IMAGE_NAME):latest
+	@$(CONTAINER_ENGINE) tag $(IMAGE_NAME):$(IMAGE_TAG) $(IMAGE_NAME):latest
+	@$(CONTAINER_ENGINE) --config=$(DOCKER_CONF) push $(IMAGE_NAME):$(IMAGE_TAG)
+	@$(CONTAINER_ENGINE) --config=$(DOCKER_CONF) push $(IMAGE_NAME):latest
 
 generate:
 	@helm lint helm/vault-manager
 	@helm template helm/vault-manager -n vault-manager -f helm/vault-manager/values-commercial.yaml > openshift/vault-manager.template.yaml
 	@helm template helm/vault-manager -n vault-manager -f helm/vault-manager/values-fedramp.yaml > openshift/vault-manager-fedramp.template.yaml
 
 build-test-container:
-	@docker build -t vault-manager-test -f tests/Dockerfile.tests .
-
-test: build-test-container
-	@docker --config=$(DOCKER_CONF) pull $(VAULT_IMAGE):$(VAULT_IMAGE_TAG)
-	@docker --config=$(DOCKER_CONF) pull $(QONTRACT_SERVER_IMAGE):$(QONTRACT_SERVER_IMAGE_TAG)
-	@docker --config=$(DOCKER_CONF) pull $(KEYCLOAK_IMAGE):$(KEYCLOAK_IMAGE_TAG)
-	@docker --config=$(DOCKER_CONF) pull $(KEYCLOAK_CLI_IMAGE):$(KEYCLOAK_CLI_IMAGE_TAG)
-	@docker run -t \
-		--rm \
-		--net=host \
-		-v $(PWD)/.env:/tests/.env \
-		-v /var/run/docker.sock:/var/run/docker.sock \
-		-e HOST_PATH=$(PWD) \
-		vault-manager-test
+	@$(CONTAINER_ENGINE) build -t $(IMAGE_NAME)-test -f tests/Dockerfile.tests .
+
+test-with-compose: build-test-container
+	@podman-compose -f $(COMPOSE_FILE) up -d --force-recreate
```

Comment:
Is podman-compose automatically installed with podman?

Reply:
No, they are separate projects. `podman-compose` is installed using pip, and our build servers have it installed.